aws/resource_aws_cognito_user_pool_test.go

```diff
@@ -35,37 +38,32 @@ func testSweepCognitoUserPools(region string) error {
 		MaxResults: aws.Int64(int64(50)),
 	}
 
-	for {
-		output, err := conn.ListUserPools(input)
-		if err != nil {
-			if testSweepSkipSweepError(err) {
-				log.Printf("[WARN] Skipping Cognito User Pool sweep for %s: %s", region, err)
-				return nil
-			}
-			return fmt.Errorf("Error retrieving Cognito User Pools: %s", err)
-		}
-
-		if len(output.UserPools) == 0 {
+	err = conn.ListUserPoolsPages(input, func(resp *cognitoidentityprovider.ListUserPoolsOutput, isLast bool) bool {
```

Comment:
I came across this available function...do we have a preference over one or the other?

Reply:
Paginators are always preferred! 😍